Compile issue without efivar?

EFIVAR is afaiu optional.
"checking for EFIVAR... no"

Yet without efivar, I get this build error:
lib/libcommon.a(libcommon_a-tpm2_eventlog_yaml.o): in function `yaml_devicepath':
tpm2-tools-5.2/lib/tpm2_eventlog_yaml.c:239: undefined reference to `efidp_format_device_path'

Problem seems to stem that headers are found but not relevant pkgconfig.
If "checking for EFIVAR" = no, shouldn't that stop the declaration of HAVE_EFIVAR_EFIVAR_H to true?

Ie, if any of the efivar tests fail, stop using efivar?

--- a/configure.ac
+++ b/configure.ac
@@ -68,8 +68,19 @@ AC_CHECK_LIB(crypto, [EVP_sm4_cfb128], [
 PKG_CHECK_MODULES([CURL], [libcurl])
 
 # pretty print of devicepath if efivar library is present
-PKG_CHECK_MODULES([EFIVAR], [efivar],,[true])
-AC_CHECK_HEADERS([efivar/efivar.h])
+# auto detect if not specified via the --with-efivar option.
+AC_ARG_WITH([efivar],
+  AS_HELP_STRING([--with-efivar], [Build with lib efivar for pretty print of device path. Default auto detect]),
+  ,
+  with_efivar=auto
+)
+
+# use the true program to avoid failing hard
+AS_IF([test "x$with_efivar" == "xauto"],
+  [PKG_CHECK_MODULES([EFIVAR], [efivar],,[true])],
+  [test "x$with_efivar" == "xyes"],
+  [PKG_CHECK_MODULES([EFIVAR], [efivar])],
+)
 
 # backwards compat with older pkg-config
 # - pull in AC_DEFUN from pkg.m4

